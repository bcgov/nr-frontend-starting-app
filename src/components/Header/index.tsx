import React from 'react';
import { Container, Grid } from '@mui/material';

import HeaderBanner from '../HeaderBanner';
import HeaderBox from '../HeaderBox';
import HeaderImage from '../HeaderImage';
import HeaderTitle from '../HeaderTitle';

const Header = () => (
  <HeaderBox>
    <Container maxWidth="lg">
      <Grid container spacing={1}>
        <HeaderBanner>
          <a href="https://gov.bc.ca">
            <HeaderImage src="https://developer.gov.bc.ca/static/BCID_H_rgb_rev-20eebe74aef7d92e02732a18b6aa6bbb.svg" alt="Go to the Government of British Columbia website" height="45px" />
          </a>
          <HeaderTitle>
            NR Sample App
          </HeaderTitle>
        </HeaderBanner>
      </Grid>
    </Container>
  </HeaderBox>
);

export default Header;
