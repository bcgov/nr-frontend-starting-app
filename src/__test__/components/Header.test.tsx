/* eslint-disable no-undef */
import { render } from '@testing-library/react';
import React from 'react';
import renderer from 'react-test-renderer';
import Header from '../../components/Header';

describe('the Header component', () => {
  it('should have the correct title', () => {
    const { getByTestId } = render(<Header />);
    expect(getByTestId('header-title').textContent).toBe('NR Sample App');
  });

  it('should match the snapshot', () => {
    const tree = renderer
      .create(<Header />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
