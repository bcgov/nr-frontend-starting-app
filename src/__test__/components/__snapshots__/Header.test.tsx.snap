// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the Header component should match the snapshot 1`] = `
<header
  className="sc-gsnTZi ftQjjz"
>
  <div
    className="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
  >
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 css-a5rdam-MuiGrid-root"
    >
      <div
        className="sc-bczRLJ jtcqdn"
      >
        <a
          href="/"
          title="Go to home page"
        >
          <img
            alt="Government of British Columbia logo"
            className="sc-dkzDqf fUaPXL"
            height="45px"
            src="https://developer.gov.bc.ca/static/BCID_H_rgb_rev-20eebe74aef7d92e02732a18b6aa6bbb.svg"
          />
        </a>
        <div
          className="sc-hKMtZM hLTdUm"
          data-testid="header-title"
        >
          NR Sample App
        </div>
      </div>
    </div>
  </div>
</header>
`;
